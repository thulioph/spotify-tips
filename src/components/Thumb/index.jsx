import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardImg, CardBody, CardTitle, CardSubtitle } from 'reactstrap';

const Thumb = ({ image, title, subTitle, handleClick }) => {
  return (
    <Card onClick={handleClick}>
      <CardImg top width="100%" src={image} alt={title} />

      <CardBody>
        <CardTitle>{title}</CardTitle>
        <CardSubtitle>{subTitle}</CardSubtitle>
      </CardBody>
    </Card>
  );
};

Thumb.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  handleClick: PropTypes.func,
};

Thumb.defaultProps = {
  image: '',
  title: '',
  subTitle: '',
  handleClick: null,
};

export default Thumb;