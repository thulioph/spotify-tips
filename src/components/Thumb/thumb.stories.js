import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Thumb from './index'

const stories = storiesOf('Thumb', module);
stories.addDecorator(withKnobs);

stories.add('full render', () => (
    <Thumb
      image={
        text('image', 'https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180')
      }
      title={
        text('title', 'Card Title')
      }
      subTitle={
        text('subTitle', 'Card SubTitle')
      }
      handleClick={() => action('handleClick')()}
    />
  ))
