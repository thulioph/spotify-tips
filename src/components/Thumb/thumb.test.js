import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Thumb from './index';

describe('Thumb component', () => {
  let component;
  let mockFunc;
  let ThumbComponent;
  
  beforeEach(() => {
    mockFunc = jest.fn();

    ThumbComponent = (
      <Thumb
        image={'my-image.png'}
        title={'My Title'}
        subTitle={'My Subtitle'}
        handleClick={mockFunc}
      />
    );

    component = mount(ThumbComponent);
  });

  it('Should render the correct snapshot', () => {
    const component = renderer.create(ThumbComponent).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Should render with the correct properties.', () => {
    expect(component.find('img').props().src).toBe('my-image.png');
    expect(component.find('.card-title').text()).toEqual('My Title');
    expect(component.find('.card-subtitle').text()).toEqual('My Subtitle');
  });

  it('Should call handleClick when the card is clicked.', () => {
    component.find('.card').simulate('click');

    expect(mockFunc).toBeCalled();
    expect(mockFunc).toBeCalledTimes(1);
  });
});