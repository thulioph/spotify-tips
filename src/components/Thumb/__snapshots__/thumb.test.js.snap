// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumb component Should render the correct snapshot 1`] = `
<div
  className="card"
  onClick={[MockFunction]}
>
  <img
    alt="My Title"
    className="card-img-top"
    src="my-image.png"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      My Title
    </h5>
    <h6
      className="card-subtitle"
    >
      My Subtitle
    </h6>
  </div>
</div>
`;
